pkg/flow/internal/controller/component.go

```diff
@@ -418,6 +421,7 @@ func (cn *ComponentNode) setRunHealth(t component.HealthType, msg string) {
 	cn.healthMut.Lock()
 	defer cn.healthMut.Unlock()
 
+	cn.cm.runningComponentTotal.WithLabelValues(cn.id.String(), t.String()).Inc()
```

Comment:
This never gets decremented so it doesn't accurately reflect the number of running components. 

Also the labels don't match up with the metric you constructed.

Reply:
Does it make more sense now?